sqlglot/dialects/fabric.py

```diff
@@ -5,6 +5,40 @@
 from sqlglot.dialects.tsql import TSQL
 
 
+def _unix_to_time_sql(self: "Fabric.Generator", expression: exp.UnixToTime) -> str:
+    """
+    Transform UnixToTime to Fabric-specific DATEADD syntax.
+
+    Fabric uses: DATEADD(MICROSECONDS, CAST(ROUND(column*1e6, 0) AS BIGINT), CAST('1970-01-01' AS DATETIME2(6)))
+    """
+    timestamp = expression.this
+
+    # Convert unix timestamp (seconds) to microseconds and round to avoid decimals
+    microseconds = exp.Cast(
+        this=exp.Round(
+            this=exp.Mul(this=timestamp, expression=exp.Literal.number("1e6")),
+            decimals=exp.Literal.number("0"),
+        ),
+        to=exp.DataType(this=exp.DataType.Type.BIGINT),
+    )
```

Comment:
Keep in mind that `UnixToTime` can have a scale, e.g. see how BigQuery [implements](https://github.com/tobymao/sqlglot/blob/8de87e3f755a40b600aa94ee2c30cf697ef7c43c/sqlglot/dialects/bigquery.py#L196-L210) its generator.

What's the motivation behind this addition? What problem (e.g., transpilation) are you trying to solve?